.github/workflows/test-python.yaml

```diff
@@ -0,0 +1,144 @@
+      slow-platforms:
+        type: string
+        default: '["jammy"]'
+        description: |
+          The platforms to run slow tests on, as a JSON array.
+      slow-python-versions:
+        required: false
+        type: string
+        default: '["3.10"]'
+        description: |
+          The python versions to run slow tests on, as a JSON array.
+
+jobs:
+  fast:
+    name: Fast tests
+    strategy:
+      matrix:
+        platform: ${{ fromJson(inputs.fast-platforms) }}
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up uv with caching
+        id: setup-uv
+        uses: astral-sh/setup-uv@v3
+        with:
+          enable-cache: true
+          cache-suffix: ${{ toJSON(matrix.platform) }}
+      - name: Set up tests
+        shell: bash
+        run: |
+          for python_version in $(echo '${{ inputs.fast-python-versions }}' | jq -r .[] | tr '\n' ' '); do
+            python_dirname=$(echo ${{ runner.temp }} | tr '\\' /)/venv_$(echo $python_version | tr . _)
+            make setup-tests UV_PROJECT_ENVIRONMENT="${python_dirname}"
+          done
+      - name: Run tests
+        shell: bash
+        run: |
+          exit_code=0
+          for python_version in $(echo '${{ inputs.fast-python-versions }}' | jq -r .[] | tr '\n' ' '); do
+            echo "::group::Python ${python_version}"
+            python_dirname=$(echo ${{ runner.temp }} | tr '\\' /)/venv_$(echo $python_version | tr . _)
+            if ! make test-coverage UV_PROJECT_ENVIRONMENT="${python_dirname}" PYTEST_ADDOPTS="-m 'not slow'"
+            then
+              exit_code=1
+              echo "::error title=TESTS FAILED::Tests failed with Python ${python_version}"
+            fi
+            echo "::endgroup::"
+          done
+          exit $exit_code
+      - name: Upload test coverage
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-fast-${{ matrix.platform }}
+          overwrite: true
+          path: |
+            ./coverage.xml
+            htmlcov/**
+  lowest:
+    name: Minimum dependencies (all tests)
+    if: ${{ inputs.lowest-python-version }} != ''
+    runs-on: jammy
+    env:
+      UV_PYTHON: ${{ inputs.lowest-python-version }}
+      UV_RESOLUTION: lowest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up uv with caching
+        id: setup-uv
+        uses: astral-sh/setup-uv@v3
+        with:
+          enable-cache: true
+          cache-suffix: ${{ matrix.platform }}
+      - name: Install tools
+        run: |
+          make setup-tests
+      - name: Run tests
+        run: |
+          make test-coverage
+      - name: Upload test coverage
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-lowest
+          overwrite: true
+          path: |
+            ./coverage.xml
+            htmlcov/**
+  slow:
```

Comment:
The only way to do this would be to have another job that checks for slow tests and disables this job if there aren't any, which doesn't feel like it would really be a benefit. Instead a library that doesn't have slow tests can just empty the platforms, as an empty matrix will be skipped.